### ISSUE TITLE
IPv6 not configured for app.netdata.cloud ?

### ISSUE THREAD

**Issue by Martin-Luther** [netdata/netdata#10070]
```tty
Netdata user: netdata
Failed to connect to https://app.netdata.cloud, return code 7
```

##### Question summary

I was trying to claim from a machine using only one IPv6 interface (No IPv4)
I can ping and download the agent from `my-netdata.io` which is IPv6 compliant.
I can not ping or claim using `app.netdata.cloud` which is not configured for IPv6, while `netdata.cloud` is.

Could my problem be related to the fact that the subdomain `app.netdata.cloud` is not configured for IPv6 ?

I have done this several times on host with IPv4 without troubles.

##### OS / Environment

Linux myhost 4.19.0-5-amd64 #1 SMP Debian 4.19.37-5 (2019-06-19) x86_64 GNU/Linux

##### Component Name
netdata-claim.sh

**Comment by Ferroin** [comment]
Yes, it’s probably due to our current lack of IPv6 support on the Cloud itself. I’ve just mentioned this as an issue to our infrastructure team.

**Comment by Martin-Luther** [comment]
Any news ?

**Comment by Martin-Luther** [comment]
Any news ?

**Comment by kaaax0815** [comment]
I have a IPv6 Machine and please provide a alternative to Github which doesnt support IPv6

**Comment by carstenjaksch** [comment]
I just tried to claim an ipv6-only node and that failed. After adding ipv4 to the node, claiming works fine.

Please consider adding ipv6 support.

**Comment by Fabi** [comment]
when

**Comment by danielslyman** [comment]
same issue here

**Comment by MrZammler** [comment]
Hi guys, sorry for the lack of replying. There have been efforts from our infra team on it, we need to do some testing and will update hopefully soon.

**Comment by Martin-Luther** [comment]
> Hi, Any update ? ;)

I have seen [this](https://manpages.debian.org/testing/netdata-core/netdata.1.en.html#i) and checked [the official documentation of Netdata daemon](https://learn.netdata.cloud/docs/agent/daemon).

It says that `netdata-core` now supports IPv6 but I did not have the chance to test it yet.

If the daemon is now able to support IPv6, this might be the same for the client right ?

If this is true for your own installed Netdata, but it is not the case yet for `app.netdata.cloud`: see [https://ready.chair6.net/?url=app.netdata.cloud](https://ready.chair6.net/?url=app.netdata.cloud). So claiming from their service will not work.

I will try to test in a few weeks to check if they have made progress.

**Comment by AxisRay** [comment]
I don't have IPv6 address, but I have same problem.
```
Netdata user: netdata
Failed to connect to https://app.netdata.cloud, return code 7
Connection attempt 1 failed. Retry in 1s.
Failed to connect to https://app.netdata.cloud, return code 7
Connection attempt 2 failed. Retry in 2s.
Failed to connect to https://app.netdata.cloud, return code 7
Connection attempt 3 failed. Retry in 3s.
grep: /var/lib/netdata/cloud.d/tmpout.txt: No such file or directory
grep: /var/lib/netdata/cloud.d/tmpout.txt: No such file or directory
Failed to claim node with the following error message:"Unknown HTTP error message"
Unable to communicate with Netdata daemon, querying config from disk instead.
Unable to communicate with Netdata daemon, querying config from disk instead.
Token: ****************
Base URL: https://app.netdata.cloud
```

**Comment by MrZammler** [comment]
Hi @AxisRay 

`return code 7` indicates from curl `CURLE_COULDNT_CONNECT`. Assuming it wasn't a temporary problem (have you retried), it could be some other network issue, i.e. a firewall or similar. Can you please check again if it continues?

**Comment by AxisRay** [comment]
@MrZammler  Thanks, I have fixed it.
The firewall denied the connection.
My fault.

**Comment by MrZammler** [comment]
No worries thanks for the follow up!

**Comment by withinboredom** [comment]
Just ran into this problem as well on ipv6-only hosting.

**Comment by kevineduardo** [comment]
2023, and people still don't use IPv6, unbelievable.

**Comment by withinboredom** [comment]
> 2023, and people still don't use IPv6, unbelievable.

To be fair, we can kinda blame k8s for this (and many hosting providers who provide only an ipv4 load balancer and/or firewalls). I watch over a k8s cluster originally built in 2019, and it only supports ipv4 because there is not a clear upgrade path to ipv6 + ipv4, dual stack. The 'official' way is to 'completely rebuild the cluster' which is a non-starter.

**Comment by ukd1** [comment]
Also getting this, and now some newer machines I want to monitor can't connect. I'm going to assume this isn't going to happen as it's been 3 years...

**Comment by foux** [comment]
Still no news on this since almost 4 years?

**Comment by Martin-Luther** [comment]
> > 2023, and people still don't use IPv6, unbelievable.
> 
> To be fair, we can kinda blame k8s for this (and many hosting providers who provide only an ipv4 load balancer and/or firewalls). I watch over a k8s cluster originally built in 2019, and it only supports ipv4 because there is not a clear upgrade path to ipv6 + ipv4, dual stack. The 'official' way is to 'completely rebuild the cluster' which is a non-starter.

I have initially tested to claim from a Kubernetes cluster that I have installed my self with IPv6 only in 2020. IPv6 works great, but could not claim from netdata because of the lack of support. K8s is not to blame, only providers that did not implement it well, and all other editors that did not want to support IPv6.

**Comment by mario-campos** [comment]
Any ETA on this?

**Comment by kevineduardo** [comment]
Guys, give up already, the cloud now has limit to 5 nodes on the free plan, and still lacks ipv6. I switched to CheckMK Selfhosted, and it’s wonderful, zero problems.

**Comment by nevotheless** [comment]
What an ice-cold closing of an issue.

**Comment by ilyam8** [comment]
Closed as completed, ipv6 is configured for app.netdata.cloud.
